.github/workflows/validate-pr-source.yml

```
@@ -0,0 +1,60 @@
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  validate-source:
+    runs-on: ubuntu-latest
+    if: github.base_ref == 'main'
+    steps:
+      - name: Validate source branch
+        run: |
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)